includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -1048,6 +1047,31 @@ public function create_intent( $order, $prepared_source ) {
 			$request['customer'] = $prepared_source->customer;
 		}
 
+		if ( isset( $full_request['statement_descriptor'] ) ) {
+			$request['statement_descriptor'] = $full_request['statement_descriptor'];
+		}
+
+		/**
+		 * Filter the return value of the WC_Payment_Gateway_CC::generate_create_intent_request.
+		 *
+		 * @since 3.1.0
+		 * @param array $request
+		 * @param WC_Order $order
+		 * @param object $source
+		 */
+		return apply_filters( 'wc_stripe_generate_create_intent_request', $request, $order, $prepared_source );
```

Comment:
What do you think about using the new `wc_stripe_generate_create_intent_request` filter to add the `setup_future_usage` parameter, instead of overloading the `generate_create_intent_request` method? Using the filter seems like a more WordPress-y way of solving this.

No big deal either way, just pointing it out because you implemented the overloading before the filter existed, but now using the filter is an option.

Reply:
I don't have a strong opinion for either options, but the way I see it right now, there are more pros to the current method than filters:

➕  I was just working on some logic from Subscriptions, which utilized a lot of filters, which connect multiple classes together, and the flow was hard to grasp. Overloading is a much clearer sign that we're taking something, done in the parent method, and building on top of it. IMO filters should be used to connect different groups of classes, but not for connecting parent and child classes.
➕ This one is already built :D
➖  Filters would've been a bit more WordPress-y in general.

Because of this (and a tiny bit of time constraints), I prefer to stick to overloading.